test/resource.cpp

```diff
@@ -27,6 +27,138 @@ struct Host2 : Host { };
   struct Sycl2 : Sycl { };
 #endif
 
+struct NotAResource {
+};
+
+#define TEST_SPECIFIC_RESOURCE_TRAIT_POSITIVE(RES, TRAIT)             \
+  static_assert(TRAIT<RES>::value, #RES " should satisfy " #TRAIT);   \
+  static_assert(TRAIT<RES&>::value, #RES "& should satisfy " #TRAIT); \
+  static_assert(TRAIT<const RES>::value,                              \
+                "const " #RES " should satisfy " #TRAIT);             \
+  static_assert(TRAIT<const RES&>::value,                             \
+                "const " #RES "& should satisfy " #TRAIT);
+
+#define TEST_SPECIFIC_RESOURCE_TRAIT_NEGATIVE(TRAIT)                   \
+  static_assert(!TRAIT<int>::value, "int should not satisfy " #TRAIT); \
+  static_assert(!TRAIT<NotAResource>::value,                           \
+                "NotAResource should not satisfy " #TRAIT);
+
+#define TEST_TRAIT_NEGATIVE_EXTRAS(TRAIT, DERIVED)                           \
+  static_assert(!TRAIT<void*>::value, "void* should not satisfy " #TRAIT);   \
+  static_assert(!TRAIT<DERIVED>::value,                                      \
+                #DERIVED " (derived type) should not satisfy " #TRAIT);
+
+#define TEST_IS_RESOURCE_POSITIVE(RES)                                        \
```

Comment:
Isn't this the same as TEST_SPECIFIC_RESOURCE_TRAIT_POSITIVE but with TRAIT hardwired to is_resource and a test for RES&& added?